pycbc/waveform/ringdown.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (C) 2016 Miriam Cabero Mueller
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""Generate ringdown templates in the frequency domain.
+"""
+
+import numpy, lal
+from pycbc.types import TimeSeries, FrequencySeries, complex128, zeros
+
+default_ringdown_args = {'t_0':0, 'phi_0':0, 'Amp':1}
+
+def props_ringdown(obj, **kwargs):
+    """ DOCUMENT ME !!
+    """
+    pr = {}
+    if obj is not None:
+        if hasattr(obj, '__dict__'):
+            pr = obj.__dict__
+        elif hasattr(obj, '__slots__'):
+            for slot in obj.__slots__:
+                if hasattr(obj, slot):
+                    pr[slot] = getattr(obj, slot)
+        else:
+            for name in dir(obj):
+                try:
+                    value = getattr(obj, name)
+                    if not name.startswith('__') and not inspect.ismethod(value):
+                        pr[name] = value
+                except:
+                    continue
+
+    # Get the parameters to generate the waveform
+    # Note that keyword arguments override values in the template object
+    input_params = default_ringdown_args.copy()
+    input_params.update(pr)
+    input_params.update(kwargs)
+
+    return input_params
+
+def qnm_time_decay(tau, decay):
+    """Return the time at which the amplitude of the 
+    ringdown falls to 1/decay of the peak amplitude.
+
+    Parameters
+    ----------
+    tau : float
+        The damping time of the sinusoid.
+    decay: float
+        The fraction of the peak amplitude.
+
+    Returns
+    -------
+    t_decay: float
+        The time at which the amplitude of the time-domain
+        ringdown is 1/decay of the peak amplitude.
+    """
+
+    t_decay = tau * numpy.log(decay)
+    return t_decay
+
+def qnm_freq_decay(f_0, tau, decay):
+    """Return the frequency at which the amplitude of the 
+    ringdown falls to 1/decay of the peak amplitude.
```

Comment:
Please make decay a fraction, not 1/fraction; I think it's more intuitive. I think that means you'll have to invert decay in the code below.